File: C++/121.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res = 0;
        int minPrices = INT_MAX;
        for (auto prices: prices) {
            res = max(res, prices-minPrices);
            minPrices = min(minPrices, prices);
        }
        return res;
    }
};

File: Python/121.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res = 0; minPrices = 99999
        for i in range(len(prices)):
            res = max(res, prices[i]-minPrices)
            minPrices = min(minPrices, prices[i])
        return res